src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -566,86 +575,84 @@ private static bool TryParseStatusLine(HttpResponseMessage response, string resp
 
             // Clear the header if status line is recieved again. This signifies that there are multiple response headers (like in redirection).
             response.Headers.Clear();
-
             response.Content.Headers.Clear();
 
-            int responseHeaderLength = responseHeader.Length;
+            int index =  HttpPrefix.Length;
+            int majorVersion = 0;
+            while(index < responseHeader.Length && char.IsDigit(responseHeader[index]))
+            {
+                majorVersion = majorVersion * 10 + (responseHeader[index] - '0');
+                index++;
+            }
 
-            // Check if line begins with HTTP/1.1 or HTTP/1.0
-            int prefixLength = HttpPrefix.Length;
-            int versionIndex = prefixLength + 2;
+            ResponseFormatAssert(majorVersion != 0);
+            ResponseFormatAssert(responseHeader[index] == '.');
+            index++;
 
-            if ((versionIndex < responseHeaderLength) && (responseHeader[prefixLength] == '1') && (responseHeader[prefixLength + 1] == '.'))
+            // Need minor version
+            ResponseFormatAssert(char.IsDigit(responseHeader[index]));
+            int minorVersion = 0;
+            while(index < responseHeader.Length && char.IsDigit(responseHeader[index]))
             {
-                response.Version =
-                    responseHeader[versionIndex] == '1' ? HttpVersion.Version11 :
-                    responseHeader[versionIndex] == '0' ? HttpVersion.Version10 :
-                    new Version(0, 0);
+                minorVersion = minorVersion * 10 + (responseHeader[index] - '0');
+                index++;
             }
-            else
+
```

Comment:
Rather than having this loop twice, consider a simple helper, e.g.
```C#
private static int ReadInt(string responseHeader, ref int index)
{
    int value = 0;
    for ( ; index < responseHeader.Length; index++)
    {
        char c = responseHeader[index];
        if (c < '0' || c > '9') break;
        value = (value * 10) + (c - '0');
    }
    return value;
}
```